In [14]:
import numpy as np
import pandas as pd
from matplotlib.pyplot import subplots
import statsmodels.api as sm
from ISLP import load_data
from ISLP.models import (ModelSpec as MS,
                         summarize,
                         poly) 

10. This question should be answered using the Carseats data set

In [19]:
Carseats = pd.read_csv('..\..\Datasets\Carseats.csv')
print(Carseats.info())
Carseats[Carseats.select_dtypes(['object']).columns] = Carseats.select_dtypes(['object']).astype('category')
print(Carseats.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sales        400 non-null    float64
 1   CompPrice    400 non-null    int64  
 2   Income       400 non-null    int64  
 3   Advertising  400 non-null    int64  
 4   Population   400 non-null    int64  
 5   Price        400 non-null    int64  
 6   ShelveLoc    400 non-null    object 
 7   Age          400 non-null    int64  
 8   Education    400 non-null    int64  
 9   Urban        400 non-null    object 
 10  US           400 non-null    object 
dtypes: float64(1), int64(7), object(3)
memory usage: 34.5+ KB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   Sales        400 non-null    float64 
 1   CompPrice    400 non-null    int64   
 2  

<>:1: SyntaxWarning: invalid escape sequence '\.'
<>:1: SyntaxWarning: invalid escape sequence '\.'
C:\Users\olive\AppData\Local\Temp\ipykernel_17868\1450385880.py:1: SyntaxWarning: invalid escape sequence '\.'
  Carseats = pd.read_csv('..\..\Datasets\Carseats.csv')


(a) Fit a multiple regression model to predict Sales using Price,
Urban, and US.

In [24]:
y = Carseats['Sales']
X = MS(['Price', 'Urban', 'US']).fit_transform(Carseats)
modelo = sm.OLS(y, X)
resultados = modelo.fit()
resultados.summary()


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.239
Model:                            OLS   Adj. R-squared:                  0.234
Method:                 Least Squares   F-statistic:                     41.52
Date:                Tue, 12 May 2026   Prob (F-statistic):           2.39e-23
Time:                        15:52:27   Log-Likelihood:                -927.66
No. Observations:                 400   AIC:                             1863.
Df Residuals:                     396   BIC:                             1879.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
intercept     13.0435      0.651     20.036      0.000      11.764      14.323
Price         -0.0545      0.005    -10.389      0.000      -0.065      -0.044
Urban[Yes]    -0.0219      0.272     -0.081      0.936      -0.556       0.512
US[Yes]        1.2006      0.259      4.635      0.000       0.691       1.710
==============================================================================
Omnibus:                        0.676   Durbin-Watson:                   1.912
Prob(Omnibus):                  0.713   Jarque-Bera (JB):                0.758
Skew:                           0.093   Prob(JB):                        0.684
Kurtosis:                       2.897   Cond. No.                         628.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

(b) Provide an interpretation of each coefficient in the model. Be
careful—some of the variables in the model are qualitative!

Intercepto 13.0435 - Valor base de venda(Sales) com Urbano = Não e US = Não

Price -0.0545 - Relação negativa, dedução de Y do produto do coeficiente com o valor de Preço

Urban -0.0219 - caso seja urbano, uma dedução desse valor de coeficiente no Y

US 1.2006 - caso seja US, aumentará o coeficiente no valor de Y

In [25]:
summarize(resultados)

,coef,std err,t,P>|t|
intercept,13.0435,0.651,20.036,0.000
Price,-0.0545,0.005,-10.389,0.000
Urban[Yes],-0.0219,0.272,-0.081,0.936
US[Yes],1.2006,0.259,4.635,0.000


(c) Write out the model in equation form, being careful to handle
the qualitative variables properly.

- ŷ = 13.0435 -0.0545 x Price -0.0219  + 1.2006
- ŷ = 13.0435 -0.0545 x Price -0.0219  
- ŷ = 13.0435 -0.0545 x Price + 1.2006
- ŷ = 13.0435 -0.0545 x Price 

(d) For which of the predictors can you reject the null hypothesis
H0 : βj =0?